model/generate.go

```diff
@@ -275,25 +270,160 @@ func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, v *Version
 	}
 
 	activatedTasksInExistingBuilds, err := addNewTasks(ctx, activationInfo, v, p, projectRef, newTVPairsForExistingVariants,
-		existingBuilds, syncAtEndOpts, g.TaskID)
+		existingBuilds, syncAtEndOpts, g.Task.Id)
 	if err != nil {
 		return errors.Wrap(err, "adding new tasks")
 	}
 
 	activatedTasksInNewBuilds, err := addNewBuilds(ctx, activationInfo, v, p, newTVPairsForNewVariants,
-		existingBuilds, syncAtEndOpts, projectRef, g.TaskID)
+		existingBuilds, syncAtEndOpts, projectRef, g.Task.Id)
 	if err != nil {
 		return errors.Wrap(err, "adding new builds")
 	}
 
 	// only want to add dependencies to activated tasks
-	if err = addDependencies(t, append(activatedTasksInExistingBuilds, activatedTasksInNewBuilds...)); err != nil {
+	if err = g.addDependencies(append(activatedTasksInExistingBuilds, activatedTasksInNewBuilds...)); err != nil {
 		return errors.Wrap(err, "adding dependencies")
 	}
 
 	return nil
 }
 
+func (g *GeneratedProject) SimulateNewDependencyGraph(v *Version, p *Project, projectRef *ProjectRef) error {
```

Comment:
added the comment.
It's true that the only use of the projectRef is the identifier. We could just pass the string, but it seems clearer to me to pass the projectRef rather than the disembodied identifier.

Reply:
That's fair -- to explain my reasoning, to me it was confusing that the dependency graph needed information from the project ref that would be user-defined so I originally thought the entire inclusion was a mistake, which is why I think a projectId field would be more clear, but it's your call.

Reply:
Okay. Another thing that made it appealing to use the projectRef was that it wasn't immediately apparent to me if I should use the projectRef's identifier or id.

Reply:
Either way 🤷‍♂️ 
I'll stick with it for now, then.